toolchains/windows-xp.toolchain.cmake

```diff
@@ -0,0 +1,42 @@
+# using MinGW-w64 with appropriate flags
+
+set(CMAKE_SYSTEM_NAME Windows)
+set(CMAKE_SYSTEM_PROCESSOR x86)
+set(CMAKE_C_COMPILER "gcc")
+set(CMAKE_CXX_COMPILER "g++")
+set(CMAKE_C_FLAGS "-DWIN32_LEAN_AND_MEAN -D_WIN32_WINNT=0x0501 -DWINVER=0x0501 ${CMAKE_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "-DWIN32_LEAN_AND_MEAN -D_WIN32_WINNT=0x0501 -DWINVER=0x0501 ${CMAKE_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
+
+# Force disable all advanced CPU features for XP compatibility
+# These must be set before CMakeLists.txt processes the options
+set(NCNN_RUNTIME_CPU OFF CACHE BOOL "Disable runtime CPU dispatch for XP compatibility" FORCE)
+set(NCNN_AVX OFF CACHE BOOL "Disable AVX for XP compatibility" FORCE)
+set(NCNN_AVX2 OFF CACHE BOOL "Disable AVX2 for XP compatibility" FORCE)
+set(NCNN_AVX512 OFF CACHE BOOL "Disable AVX512 for XP compatibility" FORCE)
+set(NCNN_AVX512VNNI OFF CACHE BOOL "Disable AVX512VNNI for XP compatibility" FORCE)
+set(NCNN_FMA OFF CACHE BOOL "Disable FMA for XP compatibility" FORCE)
+set(NCNN_F16C OFF CACHE BOOL "Disable F16C for XP compatibility" FORCE)
+set(NCNN_XOP OFF CACHE BOOL "Disable XOP for XP compatibility" FORCE)
+set(NCNN_AVXVNNI OFF CACHE BOOL "Disable AVXVNNI for XP compatibility" FORCE)
+set(NCNN_AVXVNNIINT8 OFF CACHE BOOL "Disable AVXVNNIINT8 for XP compatibility" FORCE)
+set(NCNN_AVXVNNIINT16 OFF CACHE BOOL "Disable AVXVNNIINT16 for XP compatibility" FORCE)
+set(NCNN_AVXNECONVERT OFF CACHE BOOL "Disable AVXNECONVERT for XP compatibility" FORCE)
+set(NCNN_AVX512BF16 OFF CACHE BOOL "Disable AVX512BF16 for XP compatibility" FORCE)
+set(NCNN_AVX512FP16 OFF CACHE BOOL "Disable AVX512FP16 for XP compatibility" FORCE)
+
+# Enable only basic SSE2 support
+set(NCNN_SSE2 ON CACHE BOOL "Enable SSE2 for XP compatibility" FORCE)
+set(NCNN_VULKAN OFF CACHE BOOL "Disable Vulkan for XP compatibility" FORCE)
```

Comment:
You should not set isa-related options in the toolchain, especially disabling `NCNN_RUNTIME_CPU` will lose possible optimizations such as xop

Disabling AVX will automatically disable newer instruction sets such as avx2 avx512, no need to write them all

If there are specific settings for the xp target, they should be listed in the document. I have seen that the document has pointed out the compilation parameters of `-DNCNN_VULKAN=OFF -DNCNN_SIMPLEOCV=ON -DCMAKE_BUILD_TYPE=Release`